onnx/compose.py

```diff
@@ -0,0 +1,191 @@
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import copy
+from typing import List, Tuple, Text, Optional
+from onnx import GraphProto, helper
+from onnx import TensorProto as tp
+
+
+def merge(
+        g1,  # type: GraphProto
+        g2,  # type: GraphProto
+        io_map,  # type: List[Tuple[Text, Text]]
+        name=None,  # type: Optional[Text]
+        doc_string=None,  # type: Optional[Text]
+):  # type: (...) -> GraphProto
+    """Combines two ONNX graphs into a single one.
+
+    The combined graph is defined by connecting the specified set of outputs/inputs.
+
+    Arguments:
+        g1 (GraphProto): First graph
+        g2 (GraphProto): Second graph
+        io_map (list of pairs of string): The pairs of names [(out0, in0), (out1, in1), ...]
+                                          representing outputs of the first graph and inputs of the second
+                                          to be connected
+        name (string): Optional name for the combined graph
+                       By default, the name is g1.name and g2.name joined with a double undescore delimiter
+        doc_string (string): Optional docstring for the combined graph
+                             If not provided, a default docstring with the concatenation of g1 and g2 docstrings is used
+    """
+    if type(g1) is not GraphProto:
+        raise ValueError("g1 argument is not an ONNX graph")
+    if type(g2) is not GraphProto:
+        raise ValueError("g2 argument is not an ONNX graph")
+
+    io_map_g1_outs = [io[0] for io in io_map]
+    io_map_g2_ins = [io[1] for io in io_map]
+    g1_outs = [o.name for o in g1.output]
+    g2_ins = [i.name for i in g2.input]
+
+    for g1_out_name, g2_in_name in io_map:
```

Comment:
should there be a check to verify that all the outputs from g1 have a mapping to some input of g2? Without this check it is possible that some outputs from preprocessing graphs are left unmapped. If this is OK then the resultant graph output list should be extended to include this hanging output.